### PR TITLE
SCMOD-10244: Add support for suspending jobs

### PR DESCRIPTION
**Pull request by kusumaghoshdastidar**
Jira: https://portal.digitalsafe.net/browse/SCMOD-10244

### CODE REVIEW

File: job-service/README.md [hunk 1]
```diff
@@ -1,6 +1,18 @@
...
+The variable can be set to a regular expression and can be used to control which partitions should be suspended.
+
+For example to suspend only the "tenant-acme-corp" and "tenant-acme-com" partitions it can be set to `^tenant-acme-co(?rp|m)$`.
+If a job is created for a suspended partition then rather than the job being kicked off immediately as it normally would be, the Job Service marks it as suspended and it is not eligible to run at this time.
+
+Suspended jobs can be resumed by updating the `job_task_data` table.
+```
+UPDATE public.job_task_data
+SET suspended=false
+WHERE partition_id ~ '^tenant-acme-co(?rp|m)$';
```
Comment (dermot-hardy) [inline job-service/README.md:15]: This is not closed:
https://github.com/JobService/job-service/blob/SCMOD-10244-a/job-service/README.md

File: job-service-db/src/main/resources/v3.4.0/changelog.xml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext
+                                       http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd
+                                       http://www.liquibase.org/xml/ns/dbchangelog
+                                       http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.3.xsd">
+
+    <changeSet id="update_job_task_data_add_suspended" runOnChange="true" author="Gamma Team">
```
Comment (dermot-hardy) [inline job-service-db/src/main/resources/v3.4.0/changelog.xml]: Shouldn't this be in the v3.3.0 changelog file?  Should the current v3.3.0 contents be v3.4.0?

**Review by dermot-hardy — CHANGES_REQUESTED** [review]

File: job-service-db/src/main/resources/v3.3.0/changelog.xml [hunk 3]
```diff
@@ -23,10 +23,16 @@
                                        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd
                                        http://www.liquibase.org/xml/ns/dbchangelog
                                        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.3.xsd">
-    <changeSet id="add_identity_column" runOnChange="true" author="Gamma Team">
-        <addColumn schemaName="public" tableName="job">
-            <column name="identity" type="SERIAL">
-                <constraints nullable="false" />
+
+    <changeSet id="update_job_task_data_add_suspended" runOnChange="true" author="Gamma Team">
+        <preConditions onFail="MARK_RAN">
```
Comment (dermot-hardy) [inline job-service-db/src/main/resources/v3.3.0/changelog.xml]: What is this and why is it necessary?  Why isn't this identical to [v3.3.0/changelog.xml](https://github.com/JobService/job-service/blob/v3.3.0-source/job-service-db/src/main/resources/v3.3.0/changelog.xml#L27)?

File: release-notes-3.4.0.md [hunk 4]
```diff
@@ -5,6 +5,7 @@ ${version-number}
 - **SCMOD-8484**: Propagate failures through subtasks  
 Functionality has been added that allows for the failure of a task to propagate up through any tasks that are no longer able to run because the task has failed.
 - **SCMOD-9780**: Updated images to use Java 11
+- **SCMOD-10244**: Added support for suspending jobs
```
Comment (dermot-hardy) [inline release-notes-3.4.0.md]: Not a new feature.  It was in v3.3.0.

File: job-service-db/src/main/resources/v3.3.0/changelog.xml (hunk 3 above)
Comment (kusumaghoshdastidar) [reply job-service-db/src/main/resources/v3.3.0/changelog.xml]: To skip column creation if upgrading from version 3.3 which would already have the 3.3.0 changes applied.

Comment (dermot-hardy) [reply job-service-db/src/main/resources/v3.3.0/changelog.xml]: But surely skipping the changeset if it's already been applied is default behavior for Liquibase?

File: job-service-db/src/main/resources/v3.3.0/changelog.xml [hunk 5]
```diff
@@ -23,10 +23,11 @@
                                        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd
                                        http://www.liquibase.org/xml/ns/dbchangelog
                                        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.3.xsd">
-    <changeSet id="add_identity_column" runOnChange="true" author="Gamma Team">
-        <addColumn schemaName="public" tableName="job">
-            <column name="identity" type="SERIAL">
-                <constraints nullable="false" />
+
+    <changeSet id="update_job_task_data_add_suspended" runOnChange="true" author="Gamma Team">
```
Comment (dermot-hardy) [inline job-service-db/src/main/resources/v3.3.0/changelog.xml]: It worries me that this line is still not identical to the way it is in v3.3.  Hashes are involved here so there needs to be a good reason for it not to be identical.

**Review by dermot-hardy — APPROVED** [review]

**Pull request merged**